generators/php/codegen/src/ast/core/AstNode.ts

```diff
@@ -7,15 +7,18 @@ export abstract class AstNode extends AbstractAstNode {
     /**
      * Writes the node to a string.
      */
-    public toString({
-        namespace,
-        rootNamespace,
-        customConfig
-    }: {
+    public toString(param: {
         namespace: string;
         rootNamespace: string;
         customConfig: BasePhpCustomConfigSchema;
     }): string {
+        if (param == null) {
+            throw new Error(
+                "param is required. You are likely implicitly calling toString() inside a string interpolation or concatenation."
+            );
+        }
```

Comment:
Do we really need this given the type safety of this function? If so, can we prefix the error message so it's clear it's not the user's fault?

```
throw new Error(
  "Internal error; AstNode.toString method called incorrectly."
);
```

Reply:
The problem is that there isn't type safety. We override the default `toString` method with additional parameters, but that doesn't mean TypeScript stops us from doing this:
```ts
const myVar = getAstNode();
const assignment = $`{myVar} = 10;`;
```

For string concatenation or string interpolation, our custom `toString` method is called with 0 parameters. 
The goal of the error message is to help ourselves understand the issue as the default error that would be thrown is not helpful.

Having said that, I will prefix with internal.